checkr/src/middlewares/error-middleware.ts

```diff
@@ -0,0 +1,9 @@
+import { Request, Response, NextFunction } from "express";
+import { buildResponse } from "../utils/build-response";
+import { ErrorMessages } from "../constants/messages";
+import httpStatus from "http-status";
+import { IError } from "../interfaces/error-interface";
+
+export const errorMiddleware = (err: IError, req: Request, res: Response, next: NextFunction) => {
+    return buildResponse(res, err.statusCode || httpStatus.INTERNAL_SERVER_ERROR, err.message || ErrorMessages.SOMETHING_WENT_WRONG, null);
```

Comment:
search how to extend Error in javascript to create new Error class, that can handle extra use cases, 

I think that will be more organized way to store error and send error

Reply:
Will look into it,

Reply:
I have added CustomError class.